### PR TITLE
Add module-info to the api artifacts, update to Java SE 11

### PR DESCRIPTION
**Pull request by starksm64**
This relies on the staged 2.0.1-RC1 jakarta.interceptor:jakarta.interceptor-api artifact

### CODE REVIEW

File: api/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    requires transitive jakarta.interceptor;
+    requires transitive jakarta.cdi.model;
+    requires transitive jakarta.inject;
+    requires static jakarta.el;
+    // For javadoc
+    requires static java.naming;
+    //TODO: requires static jakarta.transation;
+
+    uses jakarta.enterprise.inject.se.SeContainerInitializer;
+    uses jakarta.enterprise.inject.spi.CDIProvider;
```
Comment (Ladicek) [inline api/src/main/java/module-info.java:25]: Maybe also `jakarta.enterprise.inject.build.compatible.spi.BuildServices`? I admit I don't fully understand the service provider stuff in JPMS.

File: lang-model/src/main/java/module-info.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module jakarta.cdi.model {
```
Comment (Ladicek) [inline lang-model/src/main/java/module-info.java:1]: I'd probably call it `jakarta.cdi.lang.model`.

**Comment by Ladicek** [comment]
IIUC, this also makes `javac` emit Java 11 bytecode?

Not sure if that's a good thing or a bad thing, but it should be a conscious choice.

**Comment by starksm64** [comment]
> IIUC, this also makes `javac` emit Java 11 bytecode?
> 
> Not sure if that's a good thing or a bad thing, but it should be a conscious choice.

It is essentially required as we depend on artifacts that are being built with Java 11 (annotations 2.1), and it is the baseline Java version for all major specification versions going into EE10.

**Comment by Ladicek** [comment]
I mean, I'm fine with making it 11+ only, especially if we don't have a choice :-) But it would be good if it was highlighted in the PR title / commit message.

File: api/src/main/java/module-info.java (hunk 1 above)
Comment (starksm64) [reply api/src/main/java/module-info.java:25]: You need a uses statement for any services loaded via a ServiceLoader, so yes, we should have BuildServices uses

File: lang-model/src/main/java/module-info.java (hunk 2 above)
Comment (starksm64) [reply lang-model/src/main/java/module-info.java]: Ok, updated.

**Comment by starksm64** [comment]
Ok, added updating to Java SE 11 in PR title.

**Review by Ladicek — APPROVED** [review]
Looks fine to me, though my understanding of JPMS is virtually zero :-)

**Comment by Ladicek** [comment]
I'm keeping this open for more review.

I originally thought maybe this should be a draft because it uses an RC version of Interceptors, but we already have a non-final version of Annotations, so that's fine.

Also looking at the list of commits, I think I'd merge this using "Squash and merge".

**Review by manovotn — APPROVED** [review]
LGTM  although my understanding of JPMS is not the deepest.

**Comment by starksm64** [comment]
There still an outstanding question about moving to the Java SE 11 base, but we will go with it for the RC so that we can get feedback on potential problems.

**Pull request merged**